api/layer/tagging.go

```diff
@@ -116,10 +116,18 @@ func (n *layer) GetBucketTagging(ctx context.Context, bktInfo *data.BucketInfo)
 	}
 
 	tags, err := n.treeService.GetBucketTagging(ctx, bktInfo)
-	if err != nil && !errorsStd.Is(err, ErrNodeNotFound) {
+	if err != nil {
+		if errorsStd.Is(err, ErrNodeNotFound) {
+			return nil, s3errors.GetAPIError(s3errors.ErrBucketTaggingNotFound)
+		}
+
 		return nil, err
 	}
 
+	if len(tags) == 0 {
```

Comment:
I think you can move it before the `err` check and then save on `ErrNodeNotFound` check (`tags` will be `nil` with 0 length anyway).

Reply:
The idea doesn't save the empty/nil value to the cache. If we have `ErrNodeNotFound`, then `tags` is nil/empty, we don't need to try to save it to the cache. But simultaneously we should respond to the client with a correct error.

In this case, it is better to return the previous implementation and append-only one condition after error checks
```go
if len(tags) == 0 {
	return nil, s3errors.GetAPIError(s3errors.ErrBucketTaggingNotFound)
}
```